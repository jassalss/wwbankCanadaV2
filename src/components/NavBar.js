import React, { Component } from "react";
import { NavLink } from "react-router-dom";
class NavBar extends Component {
  state = { active: "active item" };
  render() {
    return (
      <div className="ui secondary pointing menu">
        <NavLink exact to="/" className="item">
          Home
        </NavLink>
        <NavLink to="/create-account" className="item">
          Create Account
        </NavLink>
        <NavLink to="/deposit" className="item">
          Deposit
        </NavLink>
        <NavLink to="/withdraw" className="item">
          Withdraw
        </NavLink>
        <NavLink to="/transfer" className="item">
          Transfer
        </NavLink>
        <NavLink to="/account" className="item">
          Check Balance
        </NavLink>
        <NavLink to="/allAccunts" className="item">
          Display All Accounts
        </NavLink>
      </div>
    );
  }
}

export default NavBar;
