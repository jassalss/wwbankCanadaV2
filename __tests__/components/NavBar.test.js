import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
// Component to be tested
import NavBar from "../../src/components/NavBar";
describe("Testing Nav Bar", () => {
  it("renders without crashing", () => {
    shallow(<NavBar />);
  });
});
describe("Testing Nav Bar with Snapshot", () => {
  it("matches the snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <NavBar />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
