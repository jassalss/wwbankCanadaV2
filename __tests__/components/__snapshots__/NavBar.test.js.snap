// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Nav Bar with Snapshot matches the snapshot 1`] = `
<div
  className="ui secondary pointing menu"
>
  <a
    aria-current="page"
    className="item active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Home
  </a>
  <a
    aria-current={null}
    className="item"
    href="/create-account"
    onClick={[Function]}
  >
    Create Account
  </a>
  <a
    aria-current={null}
    className="item"
    href="/deposit"
    onClick={[Function]}
  >
    Deposit
  </a>
  <a
    aria-current={null}
    className="item"
    href="/withdraw"
    onClick={[Function]}
  >
    Withdraw
  </a>
  <a
    aria-current={null}
    className="item"
    href="/transfer"
    onClick={[Function]}
  >
    Transfer
  </a>
  <a
    aria-current={null}
    className="item"
    href="/account"
    onClick={[Function]}
  >
    Check Balance
  </a>
  <a
    aria-current={null}
    className="item"
    href="/allAccunts"
    onClick={[Function]}
  >
    Display All Accounts
  </a>
</div>
`;
